source/features/previous-version.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'dom-chef';
+import * as pageDetect from 'github-url-detection';
+
+import features from '../feature-manager';
+import observe from '../helpers/selector-observer';
+import * as api from '../github-helpers/api';
+import GitHubURL from '../github-helpers/github-url';
+
+async function getPreviousCommitForFile(pathname: string): Promise<string> {
+	const {user, repository, branch, filePath} = new GitHubURL(pathname);
+	const {resource} = await api.v4(`
+		resource(url: "/${user}/${repository}/commit/${branch}") {
```

Comment:
How did you find this out? I'm trying to use v4 for another issue and can't find out to use it, if it's even possible:

https://github.com/refined-github/refined-github/blob/9d80ba25d83cf329d368a4210ed7261e24ad9657/source/features/bypass-checks.tsx#L16-L17

Example check_run_id: https://github.com/microsoft/TypeScript/pull/54006/checks?check_run_id=13000796765

Reply:
Doesn't work:

```gql
{
  resource(url: "/microsoft/TypeScript/runs/13000796765") {
    resourcePath
    __typename
    ... on CheckRun {
      detailsUrl
      name
      status
    }
  }
}
```

Some random code I found online, it seems that the "ID" that GraphQL uses is the databaseId, which is different from `check_run_id`

```gql
{
  node(id: "MDg6Q2hlY2tSdW4zNDU2NjQ5MDk4") {
    __typename
    ... on CheckRun {
      resourcePath
      permalink
      externalId
      databaseId
      id
      url
    }
  }
}
```

Reply:
@fregante Sorry, I draw a blank. 😭 
I figured out reversly. Tried same way. but not works.
```js
{
  resource(url: "/microsoft/TypeScript/pull/54006") {
    ... on PullRequest {
      commits(first: 10) {
        nodes {
          commit {
            author {
              name
            }
            checkSuites(first: 20) {
              nodes {
                checkRuns(first: 10) {
                  nodes {
                    name
                    externalId
                    resourcePath
                  }
                }
                workflowRun {
                  runNumber
                  resourcePath
                }
              }
            }
          }
        }
      }
    }
  }
}
```

You can get resourcePath with this query.
And with the `resourcePath` should return the `CheckRun` but it doesn't.
When I call v3 API with `resourcePath`'s ID, It responses well.
```sh
gh api \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  /repos/microsoft/TypeScript/check-runs/13004627613
```

Maybe the GitHub's bug. 😢

Reply:
Known issue i tried very hard too, v3 is much better with the runs

Reply:
Comment updated in [`1bf6952` (#6555)](https://github.com/refined-github/refined-github/pull/6555/commits/1bf69520dd766cced310ed5f97d9d73266111def)

/OT